sdk/core/azure-core/src/main/java/com/azure/android/core/http/interceptor/LogOptions.java

```diff
@@ -146,17 +159,26 @@ public String getApplicationId() {
      */
     public LogOptions setApplicationId(final String applicationId) {
         if (!CoreUtils.isNullOrEmpty(applicationId)) {
+            RuntimeException exception;
+
             if (applicationId.length() > MAX_APPLICATION_ID_LENGTH) {
-                throw logger
-                    .logExceptionAsError(new IllegalArgumentException("'applicationId' length cannot be greater than "
-                        + MAX_APPLICATION_ID_LENGTH));
+                exception = new IllegalArgumentException("'applicationId' length cannot be greater than "
```

Comment:
For iOS we just truncated this application ID. Should we just do the same here? I don't feel like it's worth throwing an error.

Also, regardless of whether you change this or not, we should make the limitation clear in the doc comment on L157.

Reply:
Ignore this comment (or at least apply it to the appropriate place in UserAgentInterceptor) if we remove this from LogOptions

Reply:
I stated this on a different comment, but I think we should determine how undesirable it would be to have an incomplete or incorrect Application ID to determine the course of action.

Reply:
Yep, agreed. On iOS we have exactly the same problem, except worse because we accept the various options in the constructor to `UserAgentInterceptor` and most users don't expect constructors to throw exceptions.

One challenge is that for iOS we populate the applicationId as the bundle identifier if the user doesn't supply one. If the bundle identifier is > 24 characters, then the user will get an exception without ever doing anything related to the applicationId.

For Android I would think we'd want a similar default (probably group ID + artifact ID) for applicationId, so even though this isn't being set in a constructor, it'd still be surprising. Maybe let's chat with @adrianhall and see if he has some guidance.